### PR TITLE
Fix destroy propagation

### PR DESCRIPTION
**Pull request by robincartier**
I moved the branch from my fork to the project https://github.com/wanadev/PhotonUI/pull/169

### CODE REVIEW

File: src/interactive/button.js [hunk 1]
```diff
@@ -306,6 +306,27 @@ var Button = Widget.$extend({
...
+
+    /**
+     * Destroy the widget.
+     *
+     * @method destroy
+     */
+    destroy: function () {
+        if (this.leftIconName && this.leftIcon) {
+            this.leftIcon.destroy();
+            this.leftIconName = null;
```
Comment (robincartier) [inline src/interactive/button.js:319]: @flozz I dont know if removing the name is needed.

**Review by jbghoul — APPROVED** [review]

**Review by jbghoul — APPROVED** [review]

**Comment by robincartier** [comment]
Dataview needs to be a widgets not a layout. A layout can add/remove single children but a dataView cannot

**Comment by robincartier** [comment]
@flozz I changed Dataview naming, could you take a look ? thx

File: src/container/menuitem.js [hunk 2]
```diff
@@ -196,6 +196,21 @@ var MenuItem = Container.$extend({
...
+    // ====== Public methods ======
+
+    /**
+     * Destroy the widget.
+     *
+     * @method destroy
+     */
+    destroy: function () {
+        if (this.iconName && this.icon) {
+            this.icon.destroy();
```
Comment (aardvarksaurus) [inline src/container/menuitem.js:208]: Is it possible to have `this.icon` not `null` but `this.iconName` at `null`?

File: src/dataview/dataview.js [hunk 3]
```diff
@@ -112,6 +111,44 @@ var DataView = Widget.$extend({
         this._bindEvent("drop", this.__html.container, "drop", this.__onDrop.bind(this));
     },
 
+    /**
+     * Dataview children widgets name.
```
Comment (aardvarksaurus) [inline src/dataview/dataview.js:115]:
```suggestion
     * Dataview children widgets names.
```

**Review by aardvarksaurus — CHANGES_REQUESTED** [review]

File: src/container/menuitem.js (hunk 2 above)
Comment (robincartier) [reply src/container/menuitem.js:208]: Nope, this.icon uses the name 
image

**Review by flozz — APPROVED** [review]

**Pull request merged**